lib/sql-logging/configuration.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+module SqlLogging
+  class Configuration
+    DEFAULTS = {
+      show_sql_backtrace: true,
+      show_top_sql_queries: :total_time,
+      top_sql_queries: 10,
+      backtrace_cleaner: nil
+    }.freeze
+
+    attr_accessor(*DEFAULTS.keys)
+
+    def initialize
+      DEFAULTS.each do |k, v|
+        instance_variable_set("@#{k}", v)
+      end
+    end
+
+    def show_top_sql_queries=(value)
+      validate_allowed([false, :rows, :queries, :bytes, :total_time,
+                        :median_time], value, __method__)
+    end
+
+    def backtrace_cleaner
+      @backtrace_cleaner ||= initialize_cleaner
+    end
+
+    private
+
+    def initialize_cleaner
+      Rails.backtrace_cleaner.dup.tap do |cleaner|
+        cleaner.add_silencer { |line| line =~ %r{sql-logging/lib} }
+      end
+    end
+
+    def validate_allowed(allowed_values, value, method_name)
+      return if allowed_values.include?(value)
```

Comment:
Did you mean to use the `method_name` argument? Because nothing in this method or its one caller actually sets the value if it's valid, and there is no test to assert that it's happening.